buster/completers/base.py

```diff
@@ -38,6 +38,18 @@ def text(self):
     def text(self, value: str) -> None:
         self._text = value
 
+    @property
+    def completor(self):
+        if isinstance(self._completor, str):
+            # convert str to iterator
+            self._completor = (msg for msg in self._completor)
```

Comment:
When does this happen? And if it's a str, why do you iterate over it?

Reply:
this is more for "frontend" reasons, so that a user always expects the text to come out as a generator. So in the front end, you don't need to handle different cases. It's always a generator, even for text. However, in the case of a string, it just gets "generated" in one shot